triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Cli.scala

```diff
@@ -0,0 +1,371 @@
+
+private[trigger] final case class Cli(
+    configFile: Option[File],
+    // For convenience, we allow passing DARs on startup
+    // as opposed to uploading them dynamically.
+    darPaths: List[Path],
+    address: String,
+    httpPort: Int,
+    ledgerHost: String,
+    ledgerPort: Int,
+    authInternalUri: Option[Uri],
+    authExternalUri: Option[Uri],
+    authBothUri: Option[Uri],
+    authRedirectToLogin: AuthClient.RedirectToLogin,
+    authCallbackUri: Option[Uri],
+    maxInboundMessageSize: Int,
+    minRestartInterval: FiniteDuration,
+    maxRestartInterval: FiniteDuration,
+    maxAuthCallbacks: Int,
+    authCallbackTimeout: FiniteDuration,
+    maxHttpEntityUploadSize: Long,
+    httpEntityUploadTimeout: FiniteDuration,
+    timeProviderType: TimeProviderType,
+    commandTtl: Duration,
+    init: Boolean,
+    jdbcConfig: Option[JdbcConfig],
+    portFile: Option[Path],
+    allowExistingSchema: Boolean,
+    compilerConfig: Compiler.Config,
+) extends StrictLogging {
+
+  def loadFromConfigFile: Option[Either[ConfigReaderFailures, TriggerServiceAppConf]] =
+    configFile.map(cf => ConfigSource.file(cf).load[TriggerServiceAppConf])
+
+  def loadFromCliArgs: ServiceConfig = {
+    ServiceConfig(
+      darPaths = darPaths,
+      address = address,
+      httpPort = httpPort,
+      ledgerHost = ledgerHost,
+      ledgerPort = ledgerPort,
+      authInternalUri = authInternalUri,
+      authExternalUri = authExternalUri,
+      authBothUri = authBothUri,
+      authRedirectToLogin = authRedirectToLogin,
+      authCallbackUri = authCallbackUri,
+      maxInboundMessageSize = maxInboundMessageSize,
+      minRestartInterval = minRestartInterval,
+      maxRestartInterval = maxRestartInterval,
+      maxAuthCallbacks = maxAuthCallbacks,
+      authCallbackTimeout = authCallbackTimeout,
+      maxHttpEntityUploadSize = maxHttpEntityUploadSize,
+      httpEntityUploadTimeout = httpEntityUploadTimeout,
+      timeProviderType = timeProviderType,
+      commandTtl = commandTtl,
+      init = init,
+      jdbcConfig = jdbcConfig,
+      portFile = portFile,
+      allowExistingSchema = allowExistingSchema,
+      compilerConfig = compilerConfig,
+    )
+  }
+
+  def loadConfig: Option[ServiceConfig] =
+    loadFromConfigFile.cata(
+      {
+        case Right(cfg) => Some(cfg.toServiceConfig)
+        case Left(ex) =>
+          logger.error(
+            s"Error loading trigger service config from file ${configFile}",
+            ex.prettyPrint(),
+          )
+          None
+      },
+      Some(loadFromCliArgs),
+    )
+}
+
+private[trigger] object Cli {
+
+  val DefaultHttpPort: Int = 8088
+  val DefaultMaxInboundMessageSize: Int = RunnerConfig.DefaultMaxInboundMessageSize
+  val DefaultMinRestartInterval: FiniteDuration = FiniteDuration(5, duration.SECONDS)
+  val DefaultMaxRestartInterval: FiniteDuration = FiniteDuration(60, duration.SECONDS)
+  // Adds up to ~1GB with DefaultMaxInboundMessagesSize
```

Comment:
Is this misplaced?

Reply:
its effectively a copy paste from the previous defaults defined in `ServiceConfig` , am not completely sure but I believe it is supposed to state that the `DefaultMaxAuthCallbacks` of `250` and the `DefaultMaxInboundMessageSize` of `4194304 bytes` add up to ~1GB